core/src/main/java/org/infinispan/interceptors/distribution/L1NonTxInterceptor.java

```diff
@@ -301,11 +317,15 @@ private BasicInvocationStage handleDataWriteCommand(InvocationContext ctx, DataW
       });
    }
 
-   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue)
-         throws Throwable {
+   private BasicInvocationStage removeFromLocalL1(InvocationContext ctx, DataWriteCommand command, Object returnValue) {
       if (ctx.isOriginLocal() && !cdl.localNodeIsOwner(command.getKey())) {
+         CompletableFuture<?> pendingAcks = commandAckCollector.getCollectorCompletableFuture(command.getCommandInvocationId(), false);
```

Comment:
Oops, L1NonTxInterceptor actually sends the L1 invalidations before invokeNext, after waiting for the acks, because we're on the originator :)

With the current code we could still remove the local L1 entry and request it again from another command. Even if we get the old value from the backup, the backup write will send an L1 invalidation to when it actually applies the value. However, that's only because L1NonTxInterceptor uses `ctx.getOrigin()`, which is the primary, instead of the actual originator of the write. So it's probably better to keep this wait.